src/payment/strategies/digitalriver/digitalriver-payment-strategy.ts

```diff
@@ -265,4 +268,51 @@ export default class DigitalRiverPaymentStrategy implements PaymentStrategy {
 
         return state;
     }
+
+    private async _processAdditionalAction(error: unknown, shouldSetAsDefaultInstrument: boolean, instrumentId: string, methodId: string): Promise<InternalCheckoutSelectors> {
+        if (!(error instanceof RequestError) || !some(error.body.errors, {code: 'additional_action_required'})) {
+            return Promise.reject(error);
+        }
+
+        const additionalAction: DigitalRiverAdditionalProviderData = error.body.provider_data;
+
+        if (!this._digitalRiverCheckoutData) {
+            throw new InvalidArgumentError('Unable to proceed because payload payment argument is not provided.');
+        }
+
+        return this._getDigitalRiverJs().authenticateSource({
+            sessionId: this._digitalRiverCheckoutData.sessionId,
+            sourceId: additionalAction.source_id,
+            sourceClientSecret: additionalAction.source_client_secret,
+        }).then((data: DigitalRiverAuthenticateSourceResponse) => {
+            if (data.status === AuthenticationSourceStatus.complete || data.status === AuthenticationSourceStatus.authentication_not_required) {
+                if (!this._digitalRiverCheckoutData) {
+                    throw new InvalidArgumentError('Unable to proceed because payload payment argument is not provided.');
+                }
+
+                const paymentPayload = {
+                    methodId,
+                    paymentData: {
+                        formattedPayload: {
+                            bigpay_token: {
+                                token: instrumentId,
+                            },
+                            verification_value: '000',
+                            credit_card_token: {
+                                token: JSON.stringify({
+                                    checkoutId: this._digitalRiverCheckoutData.checkoutId,
+                                }),
+                            },
+                            confirm: true,
+                        },
+                        set_as_default_stored_instrument: shouldSetAsDefaultInstrument,
```

Comment:
As far as I know `set_as_default_stored_instrument` should be part of `formattedPayload`. When we pass a `formattedPayload` to bigpay-client-js the rest of the payload should be empty.

We can see several examples of this across this repo.

Reply:
You're right, sorry for this, it slipped my mind